src/layouts/sidebar-sidecar/server.ts

```diff
@@ -132,6 +134,15 @@ export function getStaticGenerationFunctions<
 					// paragraphCustomAlerts,
 					[anchorLinks, { headings }],
 					rewriteTutorialLinksPlugin,
+					/**
+					 * Rewrite docs content links, which are authored without prefix.
+					 * For example, in Waypoint docs authors write "/docs/some-thing",
+					 * we need this to be "/waypoint/docs/some-thing" for Dev Dot.
+					 */
+					[
+						remarkPluginAdjustLinkUrls,
+						{ urlAdjustFn: productUrlAdjusters[product.slug] },
```

Comment:
This change applies docs `productUrlAdjusters` across all docs routes. Previously this remark plugin was added on a route-by-route basis (other changes in this PR remove those route-by-route additions).

We can control how URLs are adjusted on a per-product basis (and even per-base-path basis) within `./utils/product-url-adjusters` (We already had product-specific adjustments, eg for Vault's `/api` → `/api-docs` URLs).